pkg/cli/admin/upgrade/status/controlplane.go

```diff
@@ -0,0 +1,78 @@
+package status
+
+import (
+	"io"
+	"text/template"
+
+	v1 "github.com/openshift/api/config/v1"
+)
+
+type assessmentState string
+
+const (
+	assessmentStateProgressing assessmentState = "Progressing"
+	assessmentStateCompleted   assessmentState = "Completed"
+)
+
+type operators struct {
+	Total       int
+	Available   int
+	Progressing int
+	Degraded    int
+}
+
+type controlPlaneStatusDisplayData struct {
+	Assessment assessmentState
+	Completion float64
+	Operators  operators
+}
+
+func assessControlPlaneStatus(cv *v1.ClusterVersion, operators []v1.ClusterOperator) controlPlaneStatusDisplayData {
+	var displayData controlPlaneStatusDisplayData
+	var completed int
+
+	for _, operator := range operators {
+		displayData.Operators.Total++
+		for _, condition := range operator.Status.Conditions {
+			switch {
+			case condition.Type == v1.OperatorAvailable:
+				if condition.Status == v1.ConditionTrue {
+					displayData.Operators.Available++
+				}
+			case condition.Type == v1.OperatorProgressing && condition.Status == v1.ConditionTrue:
+				displayData.Operators.Progressing++
+			case condition.Type == v1.OperatorDegraded && condition.Status == v1.ConditionTrue:
+				displayData.Operators.Degraded++
+			}
+		}
+
+		for _, version := range operator.Status.Versions {
+			if version.Name == "operator" {
+				if version.Version == cv.Spec.DesiredUpdate.Version {
+					completed++
+				}
+				break
+			}
+		}
+	}
+	if completed == displayData.Operators.Total {
+		displayData.Assessment = assessmentStateCompleted
+	} else {
+		displayData.Assessment = assessmentStateProgressing
+	}
+
+	displayData.Completion = float64(completed) / float64(displayData.Operators.Total) * 100.0
+	return displayData
+}
+
+var controlPlaneStatusTemplate = template.Must(template.New("controlPlaneStatus").Parse(controlPlaneStatusTemplateRaw))
+
+func renderControlPlaneStatus(f io.Writer, displayData controlPlaneStatusDisplayData) error {
+	return controlPlaneStatusTemplate.Execute(f, displayData)
+}
+
+const controlPlaneStatusTemplateRaw = `= Control Plane =
+Assessment:      {{ .Assessment }}
+Completion:      {{ printf "%.0f" .Completion }}%
```

Comment:
To avoid confusion vs. the `Progressing` % completion (which is based on manifest counts), it's probably worth adding qualifying text here that explains that it's a ClusterOperator percentage.  Something like:

```
Completion:      {{ printf "%.0f" .Completion }}% ({{ printf %d .ClusterOperatorCompletedCount }} of {{ printf %d .ClusterOperatorManagedCount }} cluster operators are {{ printf "%s" .TargetVersion }})

Reply:
Yes! I have this discrepancy in my notes, we'll need to deal with that somehow.

Reply:
As long as we show the completion on the top line (which is manifest-based and I am increasingly convinced it is really confusing for users, because payload manifests are not a user-facing concept, we dont even say 303 of 867 of what, and the percentage resets several times during the upgrade), having two numbers will be always be confusing...

I'm not totally against being transparent and say that we count COs here, but I'm personally leaning towards making this number opaque, so that we can change its implementation if we find more precise notions of completeness (maybe really count manifests?).